### ISSUE TITLE
Future of Coding Weekly 2020/06 Week 1

### ISSUE THREAD

**Issue by marianoguerra** [marianoguerra/future-of-coding-weekly#24]
This issue collects resources for the newsletter of the year/month/week that should be in the title

If you want to contribute open the issue and paste a resource you think is worth sharing in the newsletter and comment on the issue of the week.

**Use markdown for formatting**, ideally **a contribution should have a title a link and optionally a description.**

Post helpers: https://marianoguerra.github.io/future-of-coding-weekly/helpers.html

Use comment reactions on issue comments to up vote/down vote/whatever each submission the reactions that have a clear sense of positive/negative will be considered to rank the submissions of the week.

**Comment by marianoguerra** [comment]
📝 [Bits of History, Words of Advice](https://gbracha.blogspot.com/2020/05/bits-of-history-words-of-advice.html). Via [Mariano Guerra](https://twitter.com/warianoguerra) 🧵[Conversation](https://marianoguerra.github.io/future-of-coding-weekly/history/weekly/2020/06/W1/general.html#2020-05-25T07:51:41.201Z)

> Today Smalltalk is relegated to a small niche of true believers.  Whenever two or more
> Smalltalkers gather over drinks, the question is debated: Why?

**Comment by marianoguerra** [comment]
🎥 [Interesting idea of cruising through computational structures in VR](https://twitter.com/IrvinHwang/status/1212448443962216448). Via [Prathyush](https://twitter.com/prabros) 🧵[Conversation](https://marianoguerra.github.io/future-of-coding-weekly/history/weekly/2020/06/W1/general.html#2020-05-25T11:09:22.209Z)

**Comment by marianoguerra** [comment]
📦 [A visual programming language, evaluated as a cellular automaton ](https://github.com/felko/ton-lang). Via [Prathyush](https://twitter.com/prabros) 🧵[Conversation](https://marianoguerra.github.io/future-of-coding-weekly/history/weekly/2020/06/W1/general.html#2020-05-27T10:44:01.235Z)

**Comment by marianoguerra** [comment]
📝 [CLIs are reified UIs](https://www.expressionsofchange.org/reification-of-interaction/). Via **Garth Goldwater** 🧵[Conversation](https://marianoguerra.github.io/future-of-coding-weekly/history/weekly/2020/06/W1/general.html#2020-05-27T21:59:17.241Z)

> Many of the advantages of Command Line Interfaces can be attributed to the fact that they reify the user interaction: the interaction itself is made into something that’s visible and can be interacted with.

**Comment by marianoguerra** [comment]
📢 [Apply for participating in the Codestrates v2 Programmable Game Challenge](https://www.survey-xact.dk/LinkCollector?key=HCPGJP41UK9J). Via **Clemens Klokmose** 🧵[Conversation](https://marianoguerra.github.io/future-of-coding-weekly/history/weekly/2020/06/W1/general.html#2020-05-28T10:31:56.255Z)

> I would like to share this invitation to participate on a study on a new set of developer tools for our Webstrates platform

> The challenge will run from June 10th through June 26th ending with a demo event on Zoom Friday the 26th at 15.30 CET. Sign up before June 8th at the latest.

> The programming challenge is to make a small multiplayer game where the players can change rules through programming using our Codestrates v2 development platform. 

> The programming challenge is open-ended, but expect to spend the equivalent of a couple of evenings to work on it. You can work on the challenge by yourself or in a group (all members should sign up). To participate, you are expected to have experience with Web development and JavaScript programming. We don’t expect that you have experience with game development but it’s fine if you do. We assume games to be publicly shared under Creative Commons Attribution 4.0 International (CC BY 4.0) unless otherwise is agreed with you.

**Comment by marianoguerra** [comment]
⌨️ [Sweet-expressions (t-expressions)](https://srfi.schemers.org/srfi-110/srfi-110.html). Via **S.M Mukarram Nainar** 🧵[Conversation](https://marianoguerra.github.io/future-of-coding-weekly/history/weekly/2020/06/W1/general.html#2020-05-29T00:38:03.260Z)

**Comment by marianoguerra** [comment]
📝 [Software Development as Advanced Damage Control](http://acko.net/blog/software-development-as-advanced-damage-control/). Via [Jamie Brandon](https://scattered-thoughts.net/) 🧵[Conversation](https://marianoguerra.github.io/future-of-coding-weekly/history/weekly/2020/06/W1/general.html#2020-05-29T17:55:56.270Z)

> When you look at code from a data-centric view, a lot of things start to look like stale or inconsistent data problems. I don't like using the word "cache" for this because it focuses on the negative, the absence of fresh input. The real issue is data dependencies, which are connections that must be maintained in order to present a cohesive view and cohesive behavior, derived from a changing input model.

**Comment by marianoguerra** [comment]
📝 [The Story behind The Truth: Designing a Data Model](https://ourmachinery.com/post/the-story-behind-the-truth-designing-a-data-model/). Via [Jamie Brandon](https://scattered-thoughts.net/) 🧵[Conversation](https://marianoguerra.github.io/future-of-coding-weekly/history/weekly/2020/06/W1/general.html#2020-05-29T17:59:44.271Z)

> In short, by putting a lot of responsibilities on the data model you can make the UI code for the editor a lot simpler. This is really important to us, because one of the problems we had in Bitsquid/Stingray was that we spent a lot of time on developing UI and tools. Sometimes we would spend 30 minutes to add a feature to the runtime and then a week to create a UI for it. In The Machinery we wanted to address that imbalance and make sure that we could write tool and UI code as efficiently as runtime code. https://ourmachinery.com/writing-tools-faster.html

> How Do We Fix it? Automate undo, copy/paste, etc with a well-defined data model Less busy-work Minimize and own the tech stack Make things explicit and easy to understand Avoid changing frameworks Control performance Reuse UIs and generate them automatically from data Properties, Tree, Graph, etc Don't have to create an UI for everything.

**Comment by marianoguerra** [comment]
🎥 [Mosaic, an openFrameworks based Visual Patching Creative-Coding Platform](https://mosaic.d3cod3.org/). Via **Korede Aderele** 🧵[Conversation](https://marianoguerra.github.io/future-of-coding-weekly/history/weekly/2020/06/W1/general.html#2020-05-29T23:29:25.275Z)

> This project deals with the idea of integrate/amplify human-machine communication, offering a real-time flowchart based visual interface for high level creative coding. As live-coding scripting languages offer a high level coding environment, ofxVisualProgramming and the Mosaic Project as his parent layer container, aim at a high level visual-programming environment, with embedded multi scripting languages availability (Lua, GLSL, Python, Processing and BASH(macOS & linux) ).

**Comment by marianoguerra** [comment]
🐦 [Sky squids, shields & smartbombs #MadeInDreams](https://twitter.com/ocornut/status/1266416887082221568). Via **Korede Aderele** 🧵[Conversation](https://marianoguerra.github.io/future-of-coding-weekly/history/weekly/2020/06/W1/general.html#2020-05-30T05:53:00.277Z)

> There's an interesting look coming from many contents made in Dreams, which isn't only due to the great rendering engine (and lack of hard edges) but also the process of modeling with tools tightly designed around its tech.

**Comment by marianoguerra** [comment]
💬 [I'm evaluating several different approaches to making this happen, and it would help enormously to hear use cases people are dreaming of](https://twitter.com/jryans/status/1266166240114020352). Via **J. Ryan Stinnett** 🧵[Conversation](https://marianoguerra.github.io/future-of-coding-weekly/history/weekly/2020/06/W1/feedback.html#2020-05-29T22:14:30.218Z)

> Naming specific apps and goals you want to achieve by slicing them up into frames is even better.

**Comment by marianoguerra** [comment]
💬 ['ve put together a few slides about Glance's goals and strategy. Questions and criticism welcome](https://docs.google.com/presentation/d/1vm0uyc_swsSMTe_uIigCH-TBWj3QAX_4wYFoamFgI4w/edit?usp=sharing). Via **Robbie Gleichman** 🧵[Conversation](https://marianoguerra.github.io/future-of-coding-weekly/history/weekly/2020/06/W1/feedback.html#2020-05-30T05:47:10.225Z)

> TLDR: The Glance project aims to "dramatically increase the programming capabilities of everyone (especially beginners) by creating a visual programming inspired software creation tool that is both easy to learn and powerful enough to solve complex problems."

**Comment by marianoguerra** [comment]
🎥 [this week I'll give you a short introduction about how a react-native app can be created with my flow-based-programming editor running locally](https://youtu.be/a6ZQDTl5CWI) by [Maikel van de Lisdonk](https://www.devhelpr.com/) 🧵[Conversation](https://marianoguerra.github.io/future-of-coding-weekly/history/weekly/2020/06/W1/two-minute-week.html#2020-05-25T18:00:56.440Z)

🎥 [Brief update on Glance](https://youtu.be/dU0rHwNXCig) by **Robbie Gleichman** 🧵[Conversation](https://marianoguerra.github.io/future-of-coding-weekly/history/weekly/2020/06/W1/two-minute-week.html#2020-05-25T22:10:17.442Z)

> I’ve been working on a presentation about what problem Glance is attempting to solve, why that problem is worth solving, and it proposes a plan for how we can attempt to solve it. I plan to share this presentation within the next day or two. 
> I have started investigating how to make a GUI for Glance, by which I mean a visual programming editor that will generate Haskell. Right now I’m evaluating the GUI toolkit GTK because it has Haskell bindings. The existing Glance code creates diagrams using the Haskell diagrams library, and I should be able to render these diagrams in GTK using the Cairo or GTK backends for the diagrams library.

🎥 [2-minute video of the Mu project](https://archive.org/details/akkartik-2min-2020-05-29) by [Kartik Agaram](http://akkartik.name/about) 🧵[Conversation](https://marianoguerra.github.io/future-of-coding-weekly/history/weekly/2020/06/W1/two-minute-week.html#2020-05-30T06:47:11.445Z)

🎥 [Here is my Week 7 update](https://files.slack.com/files-pri/T5TCAFTA9-F01596WEW2C/week7.mp4) by [Chris Maughan](https://twitter.com/cmaughan) 🧵[Conversation](https://marianoguerra.github.io/future-of-coding-weekly/history/weekly/2020/06/W1/two-minute-week.html#2020-05-31T09:35:05.446Z)

> In this video I mostly show the output of the synthesizer with 3 different sound setups.  Adding the effects units make things fun to play with!

> I didn't mention the technical aspects much in the video, but I've been working/thinking about:
> * How to manage the data structure to represent the timeline of note events.  I have a prototype that involves a linked list of timeline events, allocated from a memory pool which is thread safe.  It seems to work OK, but it is ongoing research.
> * Bugs!  I had a significant bug with the 'widen' button on the master output.  That just adds a small delay to the right stereo channel.  Every time I turned it on I would get audio corruption.  I eventually realized that this was caused by the spectrum analyzer(!) which was reading the graph 'too early' and breaking the 'history' buffer of the delay module.  As I also mention in the video, I have an audio glitch which could be one of several things, but is most likely caused by the new effect modules.

> The current graph layout matches a synthesizer I have for my iPad, as I've mentioned before - Audio Kit Synth One.  This is great because I can try to reproduce sound effects from that synth; but I want to move on from that and create my own layouts - since at the heart of this design is the idea that live coders can build synthesizers to match what they want to create and tweak them 'live'

**Comment by marianoguerra** [comment]
📢 [Machine Learning on Source Code](https://ml4code.github.io/). Via **Maeliza** 🧵[Conversation](https://marianoguerra.github.io/future-of-coding-weekly/history/weekly/2020/06/W1/general.html#2020-05-28T03:52:59.252Z)

> Is anyone working/ interested by ML on Code? This is a topics I've been looking into with Shubhadeep Roychowdhury. We believe it to be such a promising field for future of coding. 

> ➡ ground breaking function/docstring mismatch detection 
> ➡ pityless variable misuse 
> ➡ dope auto-complete

> Some really nice articles and pieces of code are getting out

**Comment by marianoguerra** [comment]
https://tinyletter.com/marianoguerra/letters/future-of-coding-weekly-2020-06-week-1
